src/overlay/overlay.tsx

```diff
@@ -236,14 +250,40 @@ const OverlayComponent = ({
     // =============================================================================
     // EVENT HANDLERS
     // =============================================================================
-    const handleWrapperClick = (event: React.MouseEvent<HTMLDivElement>) => {
-        const modal = childRef.current?.firstChild;
-        if (modal && (modal as Node).contains(event.target as Node)) {
+    const handleDocumentMouseDown = (e: MouseEvent) => {
+        const container = containerRef?.current || wrapperRef.current;
+        if (!container) {
+            mouseDownInsideModalRef.current = false;
             return;
-        } else if (onOverlayClick && enableOverlayClick) {
+        }
+
+        const target = e.target as Node;
+        const clickedOnContainer = container === target;
+        const clickedInsideContainer = container.contains(target);
+
+        // Track if mousedown started inside modal content
+        mouseDownInsideModalRef.current =
+            clickedInsideContainer && !clickedOnContainer;
+    };
+
+    const handleWrapperClick = (event: React.MouseEvent<HTMLDivElement>) => {
+        const wrapper = event.currentTarget;
+        const target = event.target as Node;
+        const clickedInsideModal =
+            wrapper !== target && wrapper.contains(target);
```

Comment:
I think the logic here also needs to check against the `containerRef`/`wrapperRef` (as currentTarget is always pointing to the overlay container)

could we implement the `containerRef` with Modal and ModalV2?